packages/cardpay-reward-programs/cardpay_reward_programs/rules/staking.py

```diff
@@ -0,0 +1,107 @@
+                    WHERE a.safe = b.safe
+                    AND _block_number::integer < $2::integer
+                )
+                group by safe, _block_number, type
+            ),
+            filtered_balances as (
+                select tht.safe, tht.balance_downscale_e9_uint64::int64 as balance_int64, tht._block_number, tht.token,
+                from {token_holder_table} as tht, filtered_safes as sot
+                where tht.safe = sot.safe
+                and sot.type = 'depot'
+            ),
+            balance_changes as (select
+                safe,
+                balance_int64 - lag(balance_int64, 1 ,0) over (partition by safe order by _block_number asc) as change,
+                $5::float+1 as interest_rate,
+                ($2::integer - _block_number::integer) / $4::float  as compounding_rate,
+                from filtered_balances
+                where _block_number::integer < $2::integer
+                and token = $3::text
+                and safe is not null
+                qualify
+                _block_number::integer >= $1::integer
+                ),
+            original_balances as (select safe,
+                last(balance_int64) as change,
+                $5::float+1 as interest_rate,
+                1 as compounding_rate,
+                from filtered_balances a
+                where _block_number::integer < $1::integer
+                and token = $3::text
+                and safe is not null
+                and _block_number = (
+                    SELECT MAX(_block_number)
+                    FROM filtered_balances b
+                    WHERE a.safe = b.safe
+                    AND _block_number::integer < $1::integer
+                )
+                group by safe
+                ),
+            all_data as (select * from original_balances union all select * from balance_changes)
+
+            select owner as payee, sum(change * ((interest_rate**compounding_rate) - 1)) as rewards
+            from all_data
+            left join filtered_safes using (safe)
+            group by payee
+        """
+
+    def run(self, payment_cycle: int, reward_program_id: str):
+        start_block, end_block = (
+            payment_cycle - self.payment_cycle_length,
+            payment_cycle,
+        )
+        vars = [
+            start_block,  # $1 -> int
+            end_block,  # $2 -> int
+            self.token,  # $3 -> str
+            self.payment_cycle_length,  # $4 -> int
+            self.interest_rate_monthly,  # $5 -> float
+        ]
+
+        table_query = self._get_table_query(
+            "token_holder", "token_holder", self.start_block, payment_cycle
+        )
+
+        aux_table_query = self._get_table_query(
+            "safe_owner", "safe_owner", self.start_block, payment_cycle
+        )
+
+        if table_query == "parquet_scan([])" or aux_table_query == "parquet_scan([])":
+            df = pd.DataFrame(columns=["payee", "rewards"])
```

Comment:
["payee", "rewards"]->["payee", "amount"]

Reply:
This gets swapped out on lines 105/106